core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/dao/CustomerDAO.java

```diff
@@ -0,0 +1,183 @@
+      statement.setString(2, dto.getLastName());
+      statement.setString(3, dto.getEmail());
+      statement.setString(4, dto.getPhone());
+      statement.setString(5, dto.getAddress());
+      statement.setString(6, dto.getCity());
+      statement.setString(7, dto.getState());
+      statement.setString(8, dto.getZipCode());
+      statement.execute();
+      int id = this.getLastVal(CUSTOMER_SEQUENCE);
+      return this.findById(id);
+    }catch(SQLException e){
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+  }
+
+  public List<Customer> findAllSorted(int limit){
+    List<Customer> customers = new ArrayList<>();
+    try(PreparedStatement statement = this.connection.prepareStatement(GET_ALL_LMT);){
+      statement.setInt(1, limit);
+      ResultSet rs = statement.executeQuery();
+      while(rs.next()){
+        Customer customer = new Customer();
+        customer.setId(rs.getLong("customer_id"));
+        customer.setFirstName(rs.getString("first_name"));
+        customer.setLastName(rs.getString("last_name"));
+        customer.setEmail(rs.getString("email"));
+        customer.setPhone(rs.getString("phone"));
+        customer.setAddress(rs.getString("address"));
+        customer.setCity(rs.getString("city"));
+        customer.setState(rs.getString("state"));
+        customer.setZipCode(rs.getString("zipcode"));
+        customers.add(customer);
+      }
+    }catch(SQLException e){
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+    return customers;
+  }
+
+  public List<Customer> findAllPaged(int limit, int pageNumber){
+    List<Customer> customers = new ArrayList<>();
+    int offset = ((pageNumber-1) * limit);
+    try(PreparedStatement statement = this.connection.prepareStatement(GET_ALL_PAGED);){
+      if(limit<1){
+        limit=10;
+      }
+      statement.setInt(1, limit);
+      statement.setInt(2, offset);
+      ResultSet rs = statement.executeQuery();
+      while(rs.next()){
+        Customer customer = new Customer();
+        customer.setId(rs.getLong("customer_id"));
+        customer.setFirstName(rs.getString("first_name"));
+        customer.setLastName(rs.getString("last_name"));
+        customer.setEmail(rs.getString("email"));
+        customer.setPhone(rs.getString("phone"));
+        customer.setAddress(rs.getString("address"));
+        customer.setCity(rs.getString("city"));
+        customer.setState(rs.getString("state"));
+        customer.setZipCode(rs.getString("zipcode"));
+        customers.add(customer);
+      }
+    }catch(SQLException e){
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+    return customers;
+  }
+
+  @Override
+  public void delete(long id) {
+    try(PreparedStatement statement = this.connection.prepareStatement(DELETE);){
+      statement.setLong(1, id);
+      statement.execute();
+    }catch (SQLException e){
+      e.printStackTrace();
+      throw new RuntimeException(e);
```

Comment:
never use e.printStrackTrack..the caller will catch and log the exception.

fix all similar issues